src/rgw/rgw_rest_s3.cc

```diff
@@ -6257,6 +6257,16 @@ int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t ofs, off_
 
     status = run_s3select(m_sql_query.c_str(), &(it)[0], it.length());
     if(status<0) {
+      struct rgw_http_error r;
+
+      s->formatter->open_object_section("Error");
+
+      r.http_ret = 400;
+      r.s3_code = "Error";
```

Comment:
just came across this change, i am wondering why do we need to create an object for printing these literals. why not just put
```c++
s->formatter->dump_int("status", 400 /* http return code */);
s->formatter->dump_string("Message", "Error" /* s3 code*/);
```

unless `rgw_http_error` offers a `rgw_http_error::dump()` method.

Reply:
Thank you @tchaikov. I was thinking of making use of it with continuation message(I might be wrong)